src/databricks/labs/ucx/contexts/workflow_task.py

```diff
@@ -189,7 +189,7 @@ def policies_progress(self) -> ProgressEncoder[PolicyInfo]:
         )
 
     @cached_property
-    def grants_progress(self) -> GrantProgressEncoder:
+    def grants_progress(self) -> ProgressEncoder[Grant]:
```

Comment:
Why change this type hint?

Reply:
The interface this exposes (via the type) is the encoder interface; the fact that we have a specialisation in place is an implementation detail. (It also discourages additional public methods from appearing on the specialisation.)